Add travis-ci config

anguslees opened the pull request:
Closes #19

anguslees commented:
I tried for hours to get circileci working, and travis worked first go - so I just switched to travis-ci.  Also, travis-ci lets us do matrix builds, eg we're currently testing against both go 1.6 and go 1.7.

sebgoa commented:
two things that strike me:

* why is circle CI trying to test this. (I am not seeing a circleci config in our tree)
* the api key, as long as only travis CI can decrypt it that's fine. I seem to recall we could stick those keys in env var somehow

anguslees commented:
I spent the last day or so trying to get circleci to work, before abandoning it when Travis worked first try (and has more features).  The circleci "in progress" build won't ever finish, since I removed the config and repo permissions.

The API key is encrypted and added according to the Travis ci docs recommendation.

sebgoa commented:
LGTM
